pkg/alerts/policies.go

```diff
@@ -113,19 +133,103 @@ func (alerts *Alerts) UpdatePolicy(policy Policy) (*Policy, error) {
 }
 
 // DeletePolicy deletes an existing alert policy for a given account.
-func (alerts *Alerts) DeletePolicy(id int) (*Policy, error) {
+func (a *Alerts) DeletePolicy(id int) (*Policy, error) {
 	resp := alertPolicyResponse{}
 	url := fmt.Sprintf("/alerts_policies/%d.json", id)
 
-	_, err := alerts.client.Delete(url, nil, &resp)
-
+	_, err := a.client.Delete(url, nil, &resp)
 	if err != nil {
 		return nil, err
 	}
 
 	return &resp.Policy, nil
 }
 
+func (a *Alerts) QueryPolicyCreate(accountID int, policy QueryPolicyCreateInput) (*QueryPolicy, error) {
+	vars := map[string]interface{}{
+		"accountID": accountID,
+		"policy":    policy,
+	}
+
+	resp := alertQueryPolicyCreateResponse{}
+
+	if err := a.client.Query(alertsPolicyCreatePolicy, vars, &resp); err != nil {
+		return nil, err
+	}
+
+	return &resp.QueryPolicy, nil
+}
+
+func (a *Alerts) QueryPolicyUpdate(accountID int, policyID int, policy QueryPolicyUpdateInput) (*QueryPolicy, error) {
+	vars := map[string]interface{}{
+		"accountID": accountID,
+		"policyID":  policyID,
+		"policy":    policy,
+	}
+
+	resp := alertQueryPolicyUpdateResponse{}
+
+	if err := a.client.Query(alertsPolicyUpdatePolicy, vars, &resp); err != nil {
+		return nil, err
+	}
+
+	return &resp.QueryPolicy, nil
+}
+
+// QueryPolicy queries graphql for a policy matching the given account ID and policy ID.
+func (a *Alerts) QueryPolicy(accountID, id int) (*QueryPolicy, error) {
+	policy := &QueryPolicy{}
+	policies := []*QueryPolicy{}
+	var nextCursor *string
+
+	for ok := true; ok; ok = nextCursor != nil {
+		resp := alertQueryPolicyResponse{}
+		vars := map[string]interface{}{
+			"accountID": accountID,
+			"policyIDs": []int{id},
+		}
+
+		if err := a.client.Query(alertPolicyQueryPolicy, vars, &resp); err != nil {
+			return nil, err
+		}
+
+		policies = append(policies, resp.Actor.Account.Alerts.PoliciesSearch.Policies...)
```

Comment:
This is pretty deep reference without any error checking to make sure all of that exists.

Reply:
I've replaced this with the query for just one resource to avoid the looping entirely.